src/plugins/custom/repository.ts

```diff
@@ -0,0 +1,103 @@
+import { MySQLPromisePool } from "@fastify/mysql";
+import { FastifyInstance } from "fastify";
+import fp from "fastify-plugin";
+import { RowDataPacket, ResultSetHeader } from "mysql2";
+
+export type IRepository = MySQLPromisePool & ReturnType<typeof createRepository>;
```

Comment:
I don't have strong opinions on this, if others are okay, I am too.

Reply:
+1 on @climba03003's comment, specially for simplicity; if when adopting others wants to follow that path its fine, but it will be better if we keep it simple for the demo